File: src/main/java/org/openapitools/server/model/GetShipmentResponse.java
package org.openapitools.server.model;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonValue;
import org.openapitools.server.model.ApiResponseStatusType;
import org.openapitools.server.model.ShipmentDTO;
import jakarta.validation.constraints.*;
import jakarta.validation.Valid;



public class GetShipmentResponse   {

    private ApiResponseStatusType status;
    private ShipmentDTO result;

    /**
     * Default constructor.
     */
    public GetShipmentResponse() {
    // JSON-B / Jackson
    }

    /**
     * Create GetShipmentResponse.
     *
     * @param status status
     * @param result result
     */
    public GetShipmentResponse(
        ApiResponseStatusType status, 
        ShipmentDTO result
    ) {
        this.status = status;
        this.result = result;
    }



    /**
     * Get status
     * @return status
     */
    public ApiResponseStatusType getStatus() {
        return status;
    }

    public void setStatus(ApiResponseStatusType status) {
        this.status = status;
    }

    /**
     * Get result
     * @return result
     */
    public ShipmentDTO getResult() {
        return result;
    }

    public void setResult(ShipmentDTO result) {
        this.result = result;
    }

    /**
      * Create a string representation of this pojo.
    **/
    @Override
    public String toString() {
        StringBuilder sb = new StringBuilder();
        sb.append("class GetShipmentResponse {\n");
        
        sb.append("    status: ").append(toIndentedString(status)).append("\n");
        sb.append("    result: ").append(toIndentedString(result)).append("\n");
        sb.append("}");
        return sb.toString();
    }

    /**
     * Convert the given object to string with each line indented by 4 spaces
     * (except the first line).
    */
    private static String toIndentedString(Object o) {
        if (o == null) {
          return "null";
        }
        return o.toString().replace("\n", "\n    ");
    }
}

File: lib/Models/GetShipmentResponse.cpp


#include "GetShipmentResponse.h"

using namespace Tiny;

GetShipmentResponse::GetShipmentResponse()
{
	status = ApiResponseStatusType();
	result = ShipmentDTO();
}

GetShipmentResponse::GetShipmentResponse(std::string jsonString)
{
	this->fromJson(jsonString);
}

GetShipmentResponse::~GetShipmentResponse()
{

}

void
GetShipmentResponse::fromJson(std::string jsonObj)
{
    bourne::json object = bourne::json::parse(jsonObj);

    const char *statusKey = "status";

    if(object.has_key(statusKey))
    {
        bourne::json value = object[statusKey];




        ApiResponseStatusType* obj = &status;
		obj->fromJson(value.dump());

    }

    const char *resultKey = "result";

    if(object.has_key(resultKey))
    {
        bourne::json value = object[resultKey];




        ShipmentDTO* obj = &result;
		obj->fromJson(value.dump());

    }


}

bourne::json
GetShipmentResponse::toJson()
{
    bourne::json object = bourne::json::object();






	object["status"] = getStatus().toJson();






	object["result"] = getResult().toJson();


    return object;

}

ApiResponseStatusType
GetShipmentResponse::getStatus()
{
	return status;
}

void
GetShipmentResponse::setStatus(ApiResponseStatusType  status)
{
	this->status = status;
}

ShipmentDTO
GetShipmentResponse::getResult()
{
	return result;
}

void
GetShipmentResponse::setResult(ShipmentDTO  result)
{
	this->result = result;
}



